packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -489,7 +490,7 @@ class AndroidWebViewController extends PlatformWebViewController {
   ///
   /// See https://developer.android.com/reference/android/webkit/WebChromeClient#onGeolocationPermissionsHidePrompt()
   Future<void> setGeolocationPermissionsPromptCallbacks({
-    void Function(GeolocationPermissionsRequest geolocationPermissionsRequest)?
+    Future<GeolocationPermissionsResponse> Function(String origin)?
```

Comment:
> Sorry, I just realized looking at this again that we should make the parameter an object still, so that we have the ability to add fields to it in the future. `GeolocationPermissionsRequestParams` maybe?

I added the request params.

Reply:
Returning a response object rather than using a callback method is a harder decision. Java uses classes with single methods because it doesn't support anonymous functions. So the decision is depends on whether we would expect more methods to be added. Since this native class only handles allowing or denying the prediction, I would guess that more methods aren't expected to be added. So a `Future` response works for me!